### PR TITLE
[MM-45271]: custom group admin console page

### PR DESCRIPTION
**Pull request by michelengelen**
#### Summary
- adds a new row to the system roles section of the system console.
- adds the details page with a dummy text (tbd by @cwarnermm?)

#### Ticket Link
[MM-45271](https://mattermost.atlassian.net/browse/MM-45271)

#### Related Pull Requests
- Has server changes: https://github.com/mattermost/mattermost-server/pull/20573

##### Overview page
Screenshot 2022-06-30 at 10 12 34

##### Detail page
Screenshot 2022-06-30 at 10 12 42

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by michelengelen** [comment]
@laneycs please assign UX since I do not know who is responsible for this.

Also cc @furqanmlk for future QA review (will add the QA label and review request once this is good for review)

**Comment by mkraft** [comment]
> @laneycs please assign UX since I do not know who is responsible for this.
> 
> Also cc @furqanmlk for future QA review (will add the QA label and review request once this is good for review)

I believe @matthewbirtch would do the UX review, correct me if I'm wrong though Matt.

File: components/admin_console/system_roles/system_role/system_role_permissions.tsx [hunk 1]
```diff
@@ -195,6 +195,10 @@ export default class SystemRolePermissions extends React.PureComponent<Props, St
             ...SECTIONS_BY_ROLES,
         };
 
+        if (this.props.role.name === Constants.PERMISSIONS_SYSTEM_CUSTOM_GROUP_ADMIN) {
+            return 'This is where the description for the custom group admin goes';
```
[the remark this answers is not in the document]
Comment (AshishDhama) [reply components/admin_console/system_roles/system_role/system_role_permissions.tsx:199]: Do we need to localise it as well or it’s fine.

[the remark this answers is not in the document]
Comment (mkraft) [reply components/admin_console/system_roles/system_role/system_role_permissions.tsx:199]: We should localize it.

**Comment by rmonga71** [comment]
Looks great - can you please make sure the role is called "Custom Group Manager" as opposed to "Custom Group Admin"? This is more consistent with existing language. Thanks!

**Comment by rmonga71** [comment]
Thanks for the quick change re: role title. Synced up with @cwarnermm and we're thinking for the dummy text that's still not filled that instead of each permission description being broken up we have:

"This role has permission to create, edit, and delete custom user groups by selecting **User groups** from the Products menu."

**Review by laneycs — APPROVED** [review]
@michelengelen PM reviewer should be @rmonga71 and/or @esethna :-)

**Review by rmonga71 — APPROVED** [review]
Looks good, thanks!

**Comment by mkraft** [comment]
@michelengelen Is this still Work In Progress? I didn't want to send this to @furqanmlk if it was.

**Comment by furqanmlk** [comment]
@michelengelen 
I noticed that server PR has been merged. But a user who is neither `System Admin` and nor `Custom Group Manager` can still `create`, `edit`, `delete` custom groups.
Please advise

**Comment by rmonga71** [comment]
> @michelengelen I noticed that server PR has been merged. But a user who is neither `System Admin` and nor `Custom Group Manager` can still `create`, `edit`, `delete` custom groups. Please advise

@furqanmlk this is because the permission to do so is still enabled for all System Users in the System Console. See below for instructions to test (I've verified this works as well):
1. System Admin goes to System Console > System Roles > Custom Group Manager and adds anyone who should have permission to modify custom user groups to the new Custom Group Manager role.
2. System Admin goes to System Console > Permissions > Edit Scheme and under the All Members section unchecks the Custom Groups permission.
3. Then if you login as the test user-1 you'll see that a user that is not a  `System Admin` and nor `Custom Group Manager` will not be able to manage groups

**Review by furqanmlk — APPROVED** [review]
Working as expected.

**Review by AshishDhama — APPROVED** [review]

**Pull request merged**